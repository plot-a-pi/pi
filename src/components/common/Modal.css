.Modal {
    z-index: 5;
    background-color: lightcyan;
    color: #252735;
    border: solid 10px black;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    width: 25rem;
    font-size: 16px;
    text-align: center;  
}