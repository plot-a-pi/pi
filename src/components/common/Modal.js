import React from 'react';
import styles from './Modal.css';
import PropTypes from 'prop-types';

const Modal = ({ modalInstructions, showModal, toggleModal, children }) => {

  return (
    <section>
      <div className={`${styles.Modal} ${showModal ? styles.modalShow : styles.modalHide}`}>
        <div className={styles.modalHeader}>
          <button className={styles.closeButton} onClick={toggleModal}>&times;</button>
          <p>{modalInstructions}</p>
          { children }
        </div>
      </div>
    </section>

  );
};

Modal.propTypes = {
  modalTitle: PropTypes.string.isRequired,
  modalInstructions: PropTypes.string.isRequired,
  showModal: PropTypes.bool.isRequired,
  toggleModal: PropTypes.func.isRequired
};

export default Modal;

// Add this code wherever you want the modal-opening button to live...
// import { useModal } from '../../hooks/useModal';
// const [show<MODAL_NAME>Modal, toggle[show<MODAL_NAME>Modal] = useModal();
// In the return...
{/* <Modal showModal={show<MODAL_NAME>Modal} toggleModal={toggle<MODAL_NAME>Modal} /> */}
// <button className={styles.modalButton} type='button' onClick={() => toggle<MODAL_NAME>Modal()}> ? </button>

