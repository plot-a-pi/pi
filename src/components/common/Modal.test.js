import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('Modal component', () => {
  it('renders the Modal', () => {
    const wrapper = shallow(<Modal modalInstructions={'test instructions'} showModal={true} toggleModal={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

