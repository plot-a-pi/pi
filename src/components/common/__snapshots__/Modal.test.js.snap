// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders the Modal 1`] = `
<section>
  <div
    className="Modal modalHide"
  >
    <div
      className="modalHeader"
    >
      <button
        className="closeButton"
      >
        Ã—
      </button>
      <p />
    </div>
  </div>
</section>
`;
